### DISCUSSION TITLE
Connect element to clustered element

### DISCUSSION THREAD

**Discussion by sabrael** [clientIO/joint#1863] [Q&A] [answered]
I have a compound graph, and I would like to create an edge between an element and a clustered element. Something along the lines of:

```
var node = makeElement(node);

var group = makeElement(groupNode);
var child = makeElement(childNode);
group.embed(child);

var edge = makeLink(node, group);
```

Note that I am not trying to connect a parent to a child - I'm aware that Dagre has some issues handling that. I would expect the graph to look something like this:

![cluster_link_issue](https://user-images.githubusercontent.com/32584035/198132272-934999e6-47c3-4053-a480-86ef2dc4c014.png)

However, this throws the following error:

`Uncaught TypeError: Cannot set properties of undefined (setting 'rank')`

Any  idea how I can accomplish this?

**Comment by kumilingus** [comment, marked as answer]
Duplicate of https://github.com/clientIO/joint/issues/455.

**Reply by sabrael** [reply]
The jsfiddle in [#455](https://github.com/clientIO/joint/issues/455) does not appear to work - the console gives the following error:

`"<a class='gotoLine' href='#54:17'>54:17</a> Uncaught ReferenceError: joint is not defined"`

Do you know what is needed to get this running, so I can see your workaround?

**Reply by sabrael** [reply]
Nevermind - got it working in my local. Thanks, I should be able to work with this!

**Reply by kumilingus** [reply]
Ok, I have updated the [JSFiddle](https://jsfiddle.net/kumilingus/yfbc5knv/) anyway.
